src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScoreUtils.java

```diff
@@ -0,0 +1,426 @@
+        final int minIdentityBases = (int) Math.ceil(minIdentity * numCoveredBases);
+        return !(new HostAlignmentReadFilter(minCoveredBases, minIdentityBases)).test(cigar, numMismatches);
+    }
+
+    /**
+     * Computes abundance scores and returns map from taxonomic id to scores
+     */
+    public static Map<String, PSPathogenTaxonScore> computeTaxScores(final Iterable<PSPathogenAlignmentHit> readTaxHits,
+                                                                     final PSTree tree) {
+        final Map<String, PSPathogenTaxonScore> taxIdsToScores = new HashMap<>();
+        final Set<String> invalidIds = new HashSet<>();
+        Double sum = 0.0;
+        for (final PSPathogenAlignmentHit hit : readTaxHits) {
+            final Collection<String> hitTaxIds = new ArrayList<>(hit.taxIDs);
+
+            //Find and omit hits to tax ID's not in the database
+            final Set<String> invalidHitIds = new HashSet<>(hitTaxIds.size());
+            for (final String taxid : hitTaxIds) {
+                if (!tree.hasNode(taxid) || tree.getLengthOf(taxid) == 0) {
+                    invalidHitIds.add(taxid);
+                }
+            }
+            hitTaxIds.removeAll(invalidHitIds);
+            invalidIds.addAll(invalidHitIds);
+
+            //Number of genomes hit by this read and number of mates in the tuple (1 for single, 2 for pair)
+            final int numHits = hitTaxIds.size();
+            if (numHits == 0) continue;
+
+            //Unambiguous read scores for the lowest common ancestor and its ancestors
+            final String lowestCommonAncestor = tree.getLCA(hitTaxIds);
+            final List<String> lcaPath = tree.getPathOf(lowestCommonAncestor);
+            for (final String taxId : lcaPath) {
+                getOrAddScoreInfo(taxId, taxIdsToScores, tree).unambiguous += hit.numMates;
+            }
+
+            //Scores normalized by genome length and degree of ambiguity (number of hits)
+            final Set<String> hitPathNodes = new HashSet<>(); //Set of all unique hits and ancestors
+            for (final String taxId : hitTaxIds) {
+                final Double score = SCORE_NORMALIZATION_FACTOR * hit.numMates / (numHits * tree.getLengthOf(taxId));
+                sum += score;
+                final List<String> path = tree.getPathOf(taxId);
+                hitPathNodes.addAll(path);
+                for (final String pathTaxID : path) {
+                    PSPathogenTaxonScore info = getOrAddScoreInfo(pathTaxID, taxIdsToScores, tree);
+                    info.score += score;
+                    taxIdsToScores.put(pathTaxID, info);
+                }
+            }
+
+            //"reads" score is the number of reads that COULD belong to each node i.e. an upper-bound
+            for (final String taxId : hitPathNodes) {
+                getOrAddScoreInfo(taxId, taxIdsToScores, tree).reads += hit.numMates;
+            }
+        }
+
+        //Scores normalized to 100%
```

Comment:
The tool produces "unnormalized" and "normalized" scores, the latter being the percentage version of the former. I've renamed SCORE_NORMALIZATION_FACTOR to make this less confusing.